### PR TITLE
[review] Add set and get ElementXML commands

### PR DESCRIPTION
**Pull request by dawagner**
Get command returns current element values (from main blackboard)
in XML format.

Set command Allows assigning a configurable element's settings directly in XML.
Notes:
  - Tuning mode must be on
  - In case of failure, all the subelements that have been successfully
    written along the access keep their new value
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/289%23discussion_r42403571%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/289%23discussion_r42406318%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/289%23issuecomment-149304295%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/289%23issuecomment-149591618%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%2077ae878cddd7f93128e1d3128e57d0466fbd92c5%20parameter/ConfigurableElement.cpp%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/289%23discussion_r42403571%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22line%20100%22%2C%20%22created_at%22%3A%20%222015-10-19T17%3A55%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20done%22%2C%20%22created_at%22%3A%20%222015-10-19T18%3A15%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ConfigurableElement.cpp%3AL47-77%22%7D%2C%20%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/289%23issuecomment-149304295%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20patch%20could%20be%20splited%20in%203%3A%5Cr%5Cn%20%20%20-%20rename%20of%20fromXml%20to%20structureFromXml%5Cr%5Cn%20%20%20-%20add%20a%20CParameterAccessContext%20in%20XmlParameterSerializerContext%5Cr%5Cn%20%20%20-%20Add%20set%20and%20get%20ElementXML%20commands%22%2C%20%22created_at%22%3A%20%222015-10-19T18%3A23%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22Following%20of%20%23288.%22%2C%20%22created_at%22%3A%20%222015-10-20T14%3A53%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/289#issuecomment-149304295'>General Comment</a></b>
- <a href='https://github.com/krocard'></a> This patch could be splited in 3:
- rename of fromXml to structureFromXml
- add a CParameterAccessContext in XmlParameterSerializerContext
- Add set and get ElementXML commands
- <a href='https://github.com/krocard'></a> Following of #288.
- [x] <a href='#crh-comment-Pull 77ae878cddd7f93128e1d3128e57d0466fbd92c5 parameter/ConfigurableElement.cpp 16'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/289#discussion_r42403571'>File: parameter/ConfigurableElement.cpp:L47-77</a></b>
- <a href='https://github.com/krocard'></a> line 100
- <a href='https://github.com/dawagner'></a> :+1: done

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/289?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/289?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/289'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: parameter/ConfigurableElement.cpp [hunk 1]
```diff
@@ -46,6 +47,30 @@ CConfigurableElement::~CConfigurableElement()
 {
 }
 
+bool CConfigurableElement::fromXml(const CXmlElement &xmlElement, CXmlSerializingContext &serializingContext)
+{
+    CXmlParameterSerializingContext& xmlParameterAccessContext = static_cast<CXmlParameterSerializingContext &>(serializingContext);
+    CParameterAccessContext &parameterAccessContext = xmlParameterAccessContext.getAccessContext();
+    if (parameterAccessContext.serializeSettings()) {
```
Comment (krocard) [inline parameter/ConfigurableElement.cpp]: line 100

Comment (dawagner) [inline parameter/ConfigurableElement.cpp]: :+1: done

**Comment by krocard** [comment]
This patch could be splited in 3:
   - rename of fromXml to structureFromXml
   - add a CParameterAccessContext in XmlParameterSerializerContext
   - Add set and get ElementXML commands

**Comment by krocard** [comment]
Following of #288.

**Comment by krocard** [comment]
Taken oven by #293.

**Comment by krocard** [comment]
:+1: split done in #293

**Pull request closed without merging**